rowboat/util/input.py

```diff
@@ -15,6 +15,9 @@ def parse_duration(raw, source=None, negative=False):
     if not raw:
         raise CommandError('Invalid duration')
 
+    if source == None:
```

Comment:
yes because an empty timedelta is a falsey value, my point is you should not need to pass that in anywhere.